### PR TITLE
Milestone3

### PR DESCRIPTION
**Pull request by janis-jenny**
- Make the BooksForm component stateful
- Implement the handleChange method
- Implement the handleSubmit method
- Connect the REMOVE_BOOK action to the component
- Implement the handleRemoveBook method

### CODE REVIEW

File: src/containers/Booklist.js [hunk 1]
```diff
@@ -1,34 +1,48 @@
...
-      <tr>
-        {books.map((book) => (<Book key={book.id} book={book} />)) }
-      </tr>
+      <thead>
+        <tr>
+          <th>Id</th>
+          <th>Title</th>
+          <th>Category</th>
+        </tr>
+      </thead>
```
Comment (afanimartin) [inline src/containers/Booklist.js:25]: This code already exists in the `Book` component. Please keep your code DRY.
The `BookList` could be as simple as:
```react
const BookList = (props) => {
const handleRemoveBook = (book)
 => {...}

return <Book key{book.id} book={book} handleRemoveBook={this.handleRemoveBook} />
}
```
Now the `Book` component could look like:
```react
const Book = (props) => {
render() {
    const book = this.props.book;

    return(
      <tr>
        <td>{this.props.book.id}</td>
        <td>{this.props.book.title}</td>
        <td>{this.props.book.category}</td>
        <td>
          <button onClick={() => this.props.handleBookRemove(book)}>Remove Book</button>
        </td>
      </tr>
    )
  }

}
```
Please make the necessary changes

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @janis-jenny, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/containers/Booklist.js (hunk 1 above)
Comment (janis-jenny) [reply src/containers/Booklist.js:25]: I don't think is the same. This is no stateless component so I shouldn't use this

File: src/containers/Bookform.js [hunk 2]
```diff
@@ -8,16 +23,56 @@ const BookForm = () => {
...
+      title,
+      category,
+    };
+
+    const { createbook } = props;
+    createbook(book);
+
+    setTitle('');
+    setCategory('');
+  };
```
Comment (rukundoeric) [inline src/containers/Bookform.js]: - [x] <b>REQUIRED:</b> this `handleSubmit` is not working as it is supposed to. it should be able to create a new book item, store it in the store and display it on the screen. but it is not working. kindly fix it.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @janis-jenny, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hi, @janis-jenny, 

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Your submit button is doing nothing, the book is not appearing in the list or at your Redux state, please fix this in order to move on.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by eliasecasta — APPROVED** [review]
## Status:  Approved 🟢

Hi, @janis-jenny, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by nidalaa** [comment]
Hello @janis-jenny ,

_This message is a result of the second opinion request_

Due to the bug, your request was lost and restored only today. We apologize for the inconvenience.

As I can see - you and Code Reviewers managed to solve the issue with unclear requirements.

**Pull request merged**